File: design-patterns/go/adapter/file_api/native_api.go
// native_api.go

package main

import "fmt"

type NativeApi struct {
}

func NewNativeApi() (nativeApi *NativeApi) {
	nativeApi = &NativeApi{}
	return
}

func (nativeApi *NativeApi) FetchData() (string) {
	// code to fetch data from native API

	fmt.Println("Fetching data from Native API")
	return "Data read from Native Api"
}

func (nativeApi *NativeApi) SendData(data string) {
	// code to send data to native API

	fmt.Printf("Sending data to Native API: %v", data)
}

File: design-patterns/ts/adapter/file-adapter/native-api.ts
// native-api.ts

import Api from "./api";

class NativeApi implements Api {
    fetchData(): string {
        console.log("Fetching data from Native API");
        return "Data read from Native Api";
    }

    sendData(data: string): void {
        console.log("Sending data to Native API: " + data);
    }
}

export default NativeApi;